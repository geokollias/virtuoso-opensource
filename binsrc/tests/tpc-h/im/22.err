Starting Impala Shell without Kerberos authentication
Connected to ip-10-0-1-251.ec2.internal:21000
Server version: impalad version 2.1.3-cdh5 RELEASE (build 20816e26a150d20d2c92c470aa40b342521b243e)
Query: use TPCH
Query: select cntrycode,
	count(*) as numcust,
	sum(c_acctbal) as totacctbal
from
	(
		select
			substring(c_phone, 1, 2) as cntrycode,
			c_acctbal
		from
			customer
		where
			substring(c_phone, 1, 2) in
				('13', '31', '23', '29', '30', '18', '17')
			and c_acctbal > (
				select
					avg(c_acctbal)
				from
					customer
				where
					c_acctbal > 0.00
					and substring(c_phone, 1, 2) in
						('13', '31', '23', '29', '30', '18', '17')
			)
			and not exists (
				select
					*
				from
					orders
				where
					o_custkey = c_custkey
			)
	) as custsale
group by
	cntrycode
order by
	cntrycode
Fetched 7 row(s) in 37.42s
