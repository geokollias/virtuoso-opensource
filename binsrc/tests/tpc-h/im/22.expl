[?1034h+---------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                |
+---------------------------------------------------------------------------------------------------------------+
| Estimated Per-Host Requirements: Memory=925.51MB VCores=2                                                     |
| WARNING: The following tables are missing relevant table and/or column statistics.                            |
| tpch.customer, tpch.orders                                                                                    |
|                                                                                                               |
| 15:MERGING-EXCHANGE [UNPARTITIONED]                                                                           |
| |  order by: cntrycode ASC                                                                                    |
| |  hosts=1 per-host-mem=unavailable                                                                           |
| |  tuple-ids=9 row-size=32B cardinality=150000                                                                |
| |                                                                                                             |
| 07:SORT                                                                                                       |
| |  order by: cntrycode ASC                                                                                    |
| |  hosts=1 per-host-mem=16.00MB                                                                               |
| |  tuple-ids=9 row-size=32B cardinality=150000                                                                |
| |                                                                                                             |
| 14:AGGREGATE [FINALIZE]                                                                                       |
| |  output: count:merge(*), sum:merge(c_acctbal)                                                               |
| |  group by: cntrycode                                                                                        |
| |  hosts=1 per-host-mem=128.00MB                                                                              |
| |  tuple-ids=8 row-size=32B cardinality=150000                                                                |
| |                                                                                                             |
| 13:EXCHANGE [HASH(cntrycode)]                                                                                 |
| |  hosts=1 per-host-mem=0B                                                                                    |
| |  tuple-ids=8 row-size=32B cardinality=150000                                                                |
| |                                                                                                             |
| 06:AGGREGATE                                                                                                  |
| |  output: count(*), sum(c_acctbal)                                                                           |
| |  group by: substring(c_phone, 1, 2)                                                                         |
| |  hosts=1 per-host-mem=128.00MB                                                                              |
| |  tuple-ids=8 row-size=32B cardinality=150000                                                                |
| |                                                                                                             |
| 05:HASH JOIN [RIGHT ANTI JOIN, PARTITIONED]                                                                   |
| |  hash predicates: o_custkey = c_custkey                                                                     |
| |  hosts=1 per-host-mem=5.51MB                                                                                |
| |  tuple-ids=0,3 row-size=35B cardinality=150000                                                              |
| |                                                                                                             |
| |--12:EXCHANGE [HASH(c_custkey)]                                                                              |
| |  |  hosts=1 per-host-mem=0B                                                                                 |
| |  |  tuple-ids=0,3 row-size=35B cardinality=150000                                                           |
| |  |                                                                                                          |
| |  04:CROSS JOIN [BROADCAST]                                                                                  |
| |  |  predicates: c_acctbal > avg(c_acctbal)                                                                  |
| |  |  hosts=1 per-host-mem=8B                                                                                 |
| |  |  tuple-ids=0,3 row-size=35B cardinality=150000                                                           |
| |  |                                                                                                          |
| |  |--10:EXCHANGE [BROADCAST]                                                                                 |
| |  |  |  hosts=1 per-host-mem=0B                                                                              |
| |  |  |  tuple-ids=3 row-size=8B cardinality=1                                                                |
| |  |  |                                                                                                       |
| |  |  09:AGGREGATE [FINALIZE]                                                                                 |
| |  |  |  output: avg:merge(c_acctbal)                                                                         |
| |  |  |  hosts=1 per-host-mem=unavailable                                                                     |
| |  |  |  tuple-ids=3 row-size=8B cardinality=1                                                                |
| |  |  |                                                                                                       |
| |  |  08:EXCHANGE [UNPARTITIONED]                                                                             |
| |  |  |  hosts=1 per-host-mem=unavailable                                                                     |
| |  |  |  tuple-ids=2 row-size=8B cardinality=1                                                                |
| |  |  |                                                                                                       |
| |  |  02:AGGREGATE                                                                                            |
| |  |  |  output: avg(c_acctbal)                                                                               |
| |  |  |  hosts=1 per-host-mem=10.00MB                                                                         |
| |  |  |  tuple-ids=2 row-size=8B cardinality=1                                                                |
| |  |  |                                                                                                       |
| |  |  01:SCAN HDFS [tpch.customer, RANDOM]                                                                    |
| |  |     partitions=1/1 files=5 size=1.18GB                                                                   |
| |  |     predicates: c_acctbal > 0.00, substring(c_phone, 1, 2) IN ('13', '31', '23', '29', '30', '18', '17') |
| |  |     table stats: 15000000 rows total                                                                     |
| |  |     columns missing stats: c_phone                                                                       |
| |  |     hosts=1 per-host-mem=176.00MB                                                                        |
| |  |     tuple-ids=1 row-size=23B cardinality=150000                                                          |
| |  |                                                                                                          |
| |  00:SCAN HDFS [tpch.customer, RANDOM]                                                                       |
| |     partitions=1/1 files=5 size=1.18GB                                                                      |
| |     predicates: substring(c_phone, 1, 2) IN ('13', '31', '23', '29', '30', '18', '17')                      |
| |     table stats: 15000000 rows total                                                                        |
| |     columns missing stats: c_phone                                                                          |
| |     hosts=1 per-host-mem=264.00MB                                                                           |
| |     tuple-ids=0 row-size=27B cardinality=1500000                                                            |
| |                                                                                                             |
| 11:EXCHANGE [HASH(o_custkey)]                                                                                 |
| |  hosts=1 per-host-mem=0B                                                                                    |
| |  tuple-ids=5 row-size=4B cardinality=150000000                                                              |
| |                                                                                                             |
| 03:SCAN HDFS [tpch.orders, RANDOM]                                                                            |
|    partitions=1/1 files=25 size=6.02GB                                                                        |
|    table stats: 150000000 rows total                                                                          |
|    columns missing stats: o_orderstatus, o_orderpriority, o_clerk, o_comment                                  |
|    hosts=1 per-host-mem=792.00MB                                                                              |
|    tuple-ids=5 row-size=4B cardinality=150000000                                                              |
+---------------------------------------------------------------------------------------------------------------+
