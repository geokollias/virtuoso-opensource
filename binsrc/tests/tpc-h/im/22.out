[?1034h+-----------+---------+-------------------+
| cntrycode | numcust | totacctbal        |
+-----------+---------+-------------------+
| 13        | 90748   | 679530260.8000009 |
| 17        | 91314   | 685030261.9100009 |
| 18        | 91293   | 684593250.4199986 |
| 23        | 90359   | 677680422.769999  |
| 29        | 91124   | 683683023.6499949 |
| 30        | 91358   | 685118386.8300066 |
| 31        | 90553   | 678671645.3500047 |
+-----------+---------+-------------------+
